sycl/include/CL/sycl/handler.hpp

```diff
@@ -792,64 +671,82 @@ class handler {
 #endif // __SYCL_DEVICE_ONLY__
   }
 
-  // single_task version with a kernel represented as a sycl::kernel.
-  // The kernel invocation method has no functors and cannot be called on host.
+  /// Invokes a SYCL kernel.
+  ///
+  /// Executes exactly once. The kernel invocation method has no functors and
+  /// cannot be called on host.
+  ///
+  /// @param SyclKernel is a SYCL kernel object.
   void single_task(kernel SyclKernel) {
     throwIfActionIsCreated();
     verifySyclKernelInvoc(SyclKernel);
     MNDRDesc.set(range<1>{1});
-    MSyclKernel = detail::getSyclObjImpl(std::move(SyclKernel));
+    MSyclKernel = unique_ptr_class<kernel>(new kernel(std::move(SyclKernel)));
```

Comment:
Is there a requirement to release MSyclKernel upon return from single_task?

Reply:
I don't think so.